contracts/epochs/FlowEpoch.cdc

```diff
@@ -364,11 +367,14 @@ pub contract FlowEpoch {
                     let currentBlock = getCurrentBlock()
                     let currentEpochMetadata = FlowEpoch.getEpochMetadata(FlowEpoch.currentEpochCounter)!
                     if currentBlock.view >= currentEpochMetadata.endView {
+                        let totalRewards = FlowIDTableStaking.epochTokenPayout
```

Comment:
This diff addresses the functional problem, I wonder if we could improve the structure at the same time. To me the underlying problem is that we are relying on a contract-global variable (`epochTokenPayout`) in two different functions. We are also handling `rewardBreakdown` and `totalRewards` separately (one is passed as an argument, one is stored in a struct) when it is critical that they are mutually consistent. 

Could we modify the return value of `calculateRewards` to return a structure that includes the total reward payout? That way we could read the global variable `epochTokenPayout` in only one place in the flow, and set/store the `rewardBreakdown` and `totalRewards` in the same place. To me, this would less prone to the kind of problem this PR addresses. Let me know what you think.

Reply:
yep, I'll modify that to use it. Unfortunately, we can't change the epoch metadata struct so we'll still have to use the rewards breakdown array for the event and metadata storage though